src/testshade/rs_simplerend.cpp

```diff
@@ -179,29 +179,199 @@ rs_transform_points(OSL::OpaqueExecContextPtr /*ec*/, OSL::ustringhash /*from*/,
     return false;
 }
 
-OSL_RSOP bool
+OSL_RSOP OSL_HOSTDEVICE bool
+rs_texture(OSL::OpaqueExecContextPtr ec, OSL::ustringhash filename,
+           OSL::TextureSystem::TextureHandle* texture_handle,
+           OSL::TextureSystem::Perthread* texture_thread_info,
+           OSL::TextureOpt& options, float s, float t, float dsdx, float dtdx,
+           float dsdy, float dtdy, int nchannels, float* result,
+           float* dresultds, float* dresultdt, OSL::ustringhash* errormessage)
+{
+#ifndef __CUDA_ARCH__
```

Comment:
Can we see if a OSL_HOST_COMPILING or other compiling flag so that casts to ShaderGlobals and accesss to RendererServices only happen on host bitcode.